src/tree-internal.component.ts

```diff
@@ -20,7 +20,7 @@ import { Observable } from 'rxjs';
         [nodeDraggable]="element"
         [tree]="tree">
 
-        <div class="folding" (click)="tree.switchFoldingType()" [ngClass]="tree.foldingType.cssClass"></div>
+        <div class="folding" (click)="tree.switchFoldingType(treeService)" [ngClass]="tree.foldingType.cssClass"></div>
```

Comment:
@lovery It would be better to keep model logic separate from services. We can create method called `switchFoldingType` in InternalTreeComponent itself and there call `tree.switchFoldingType` and fire event. This way concerns are separated and it is easier to test `Tree` without any services